src/javascript/node-package.ts

```diff
@@ -849,6 +854,28 @@ export class NodePackage extends Component {
     return lazy as unknown as string;
   }
 
+  /**
+   * Attempt to resolve the currently installed version for a given dependency.
+   *
+   * @remarks
+   * This method will first look through the current project's dependencies.
+   * If found and semantically valid (not '*'), that will be used.
+   * Otherwise, it will fall back to locating a `package.json` manifest for the dependency
+   * through node's internal resolution reading the version from there.
+   *
+   * @param dependencyName Dependency to resolve for.
+   */
+  public tryResolveDependencyVersion(dependencyName: string) {
+    try {
+      const fromDeps = this.project.deps.tryGetDependency(dependencyName);
+      const version = semver.coerce(fromDeps?.version, { loose: true });
+      if (version) return version.format();
```

Comment:
I should really fix the eslint rule for that 🙈 
```suggestion
      if (version) {
        return version.format();
      }
```

Reply:
Not a fan? :smile: 
Personally, I find it a bit cleaner, but I can definitely see where others may not feel the same way.